packages/common/src/core/schemas/internal/getLocationOptions.ts

```diff
@@ -0,0 +1,67 @@
+import { extentToBBox, getGeographicOrgExtent } from "../../../extent";
+import { IHubRequestOptions } from "../../../types";
+import { getTypeFromEntity } from "../../getTypeFromEntity";
+import { HubEntity } from "../../types/HubEntity";
+import { HubEntityType } from "../../types/HubEntityType";
+import { IHubLocation, IHubLocationOption } from "../../types/IHubLocation";
+
+/**
+ * Construct the dynamic location picker options with the entity's
+ * configured location using the following logic:
+ *
+ * 1. If there is no entity (e.g. we're creating an entity),
+ * select the "custom" option by default
+ * 2. Else if there is no location set on the entity, select the
+ * "none" option
+ * 3. Else if the entity has a location and it's the same as the
+ * current option, select it and update the option with the entity's
+ * location
+ */
+export async function getLocationOptions(
+  entity: HubEntity,
+  portalName: string,
+  hubRequestOptions: IHubRequestOptions
+): Promise<IHubLocationOption[]> {
+  const defaultExtent = await getGeographicOrgExtent(hubRequestOptions);
+  const location: IHubLocation = entity.location;
+
+  return (
+    [
+      {
+        label: "{{shared.fields.location.none:translate}}",
+        location: { type: "none" },
+      },
+      {
+        label: "{{shared.fields.location.org:translate}}",
+        description: portalName,
+        location: {
+          type: "org",
+          extent: extentToBBox(defaultExtent),
+          spatialReference: defaultExtent.spatialReference,
+        },
+      },
+      {
+        label: "{{shared.fields.location.custom:translate}}",
+        description: "{{shared.fields.location.customDescription:translate}}",
+        entityType: getTypeFromEntity(entity),
+        location: {
+          type: "custom",
+          spatialReference: defaultExtent.spatialReference,
+        },
+      },
+    ] as IHubLocationOption[]
+  ).map((option) => {
+    // TODO: Verify logic with Ben as code ^^ fails if entity is null
+    // if (!entity && option.location.type === "custom") {
```

Comment:
This is needed for the scenario 1 in the description - the creation workflow (where entity is `null`). If you make line 26 have optional chaining (`entity?.location`), I believe this should all still work?

Reply:
Ah - I've eliminated the code path where the entity is `null`. 

When we create a new entity, in order to create the `EntiyEditor` instance, I'm using this..
```
const entity = this.entity || this.getDefaultEntity(this.entityType);
```
Which calls this to construct the minimum viable entity...
```
  private getDefaultEntity(entityType: string): HubEntity {
    let entity;
    switch (entityType) {
      case 'project':
        entity = { type: 'Hub Project' } as HubEntity;
        break;
    }
    return entity;

```
So there is always an entity object - even if it's filled with defaults.

Now that i understand the intent, I'll check for `.id` and set to default if that's not present.